plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/BaseJdbcConfig.java

```diff
@@ -90,4 +92,18 @@ public BaseJdbcConfig setCacheMissing(boolean cacheMissing)
         this.cacheMissing = cacheMissing;
         return this;
     }
+
+    @Min(1)
+    public long getCacheMaximumSize()
+    {
+        return cacheMaximumSize;
+    }
+
+    @Config("metadata.cache-maximum-size")
+    @ConfigDescription("Maximum number of objects stored in the metadata cache")
+    public BaseJdbcConfig setCacheMaximumSize(long cacheMaximumSize)
+    {
+        this.cacheMaximumSize = cacheMaximumSize;
```

Comment:
do we want to reject configuration like

```
metadata.cache-maximum-size = 100
# metadata.cache-ttl is not set
```

?

Reply:
Yes, that is a good idea. As an admin, it would be nice to know that changing the cache size does not enable the cache unless `cache-til` is set. 

I added a validation method that validates the cache config to do this.